parm/config/config.fv3

```diff
@@ -61,6 +67,11 @@ case $case_in in
         export layout_y=4
         export layout_x_gfs=6
         export layout_y_gfs=4
+        if [[ "$machine" == "S4" ]]; then
```

Comment:
I'd rather not add all these machine-specific layouts if we can at all help it. It adds a lot of ugly code for a machine that isn't even a tier-1 supported machine. I know S4 is small, but is there a reason S4 can't just use the same layout as the other machines for most of these?

Besides that, this script shouldn't be looking at $CASE_ENKF. First, there is little reason the layout should be different for EnKF than for any other forecast at the same resolution. Second, this construct will not behave as you want if EnKF were ever run at the same resolution as deterministic.

Reply:
Most of these changes are here to improve cycle completion times and minimize resource usage on the system so it isn't too bogged down.  In particular, the EnKF is problematic with the default layouts.  S4 is restricted to 40 nodes per user.  Each efcs instance, by default, would use 6 nodes (i.e. 6 instances at once).  With the 4x4 tiling, it is possible to push 10 instances through at a time.  In timing experiments with 80 ensembles, this resulted in mean cycle timings of ~1.5 hours (5.5 vs 4) when running at C384/C192.  Similar speedups were observed for C192/C96.

That said, perhaps this could be addressed a different way.  I could try modifying the S4.env file to specify the layouts to use, and rewrite the layouts in config.fv3 to accept a predefined value, e.g.
`export layout_x=${layout_x:-6}`.  I'm not sure off the top of my head if this would work with the setup scripts.

Alternatively, I may be able to add some magic to setup_expt.py and config.base to specify layouts, but this could get messy, too.

If you have another suggestion, I would be willing to try it.

Reply:
Adding something to S4.env will not work since it is not loaded by setup_expt.py.

Looking through the scripts and configs, I see one other relatively clean option:

1. Add an S4-specific if-block to config.efcs that exports `layout_x` for different `CASE_ENKF`s before sourcing config.fv3.
2. As discussed above, modify the layout assignments in config.fv3 (`export layout_x=${layout_x:-6}`).

@WalterKolczynski-NOAA Would you prefer this solution?

Reply:
Yes, I think I'd like putting it in config.efcs better.

Reply:
Done.  I will test that this works on Orion and run single gdasefcs and gdasfcst jobs at 384, 192, and 96 on both Orion and S4.